File: 0002-add-two-numbers/0002-add-two-numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        carry = 0
        res = ListNode(0)
        p = res
        while l1!=None or l2!=None:
            sum = carry
            
            if l1:
                sum += l1.val
                l1 = l1.next

            if l2: 
                sum += l2.val
                l2 = l2.next

 
            print(sum)
            res.next = ListNode(sum%10)
            carry = int(sum/10)
            print(carry)
            res = res.next
        if carry:
            res.next = ListNode(carry)

        return p.next
        

File: 0002-add-two-numbers/0002-add-two-numbers.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */
var addTwoNumbers = function(l1, l2) {
    let res = new ListNode()
    let sum = 0, carry = 0
    let p=res
    while(l1!==null || l2!==null) {
        sum = (l1 ? l1.val: 0) + (l2 ? l2.val: 0) + carry
        carry =  parseInt(sum/10)
        res.next = new ListNode(sum%10)
        l1 = l1  ? l1.next : null
        l2 = l2 ? l2.next : null
        res = res.next
    }
    if(carry)
        res.next = new ListNode(carry)
    return p.next
};